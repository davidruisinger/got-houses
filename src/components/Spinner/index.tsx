import React from 'react'
import styled, { keyframes, css } from '../_utils/styledComponents'
import FlexBox from '../FlexBox'

const DOT_SIZE = 6

const bounceAnimation = () =>
  keyframes`
    0% {
      transform: translateY(-${DOT_SIZE}px); 
    }

    50% {
      transform: translateY(${DOT_SIZE}px);
    }

    100% {
      transform: translateY(-${DOT_SIZE}px);    }
  `

interface DotProps {
  delay: string
}

const Wrapper = styled(FlexBox)`
  height: calc(${DOT_SIZE}px * 3);
`

const Dot = styled.div<DotProps>`
  background-color: ${props => props.theme.ancient_ivory};
  border-radius: 50%;
  width: ${DOT_SIZE}px;
  height: ${DOT_SIZE}px;
  margin: 0 ${DOT_SIZE / 2}px;
  /* Animation */
  animation: ${() =>
    css`
      ${bounceAnimation()} 1s ease-in-out infinite;
    `};
  animation-delay: ${props => props.delay};
`

const Spinner = () => (
  <Wrapper direction="row" justifyContent="center" alignItems="center">
    <Dot delay="0s" />
    <Dot delay="-0.9s" />
    <Dot delay="-0.8s" />
  </Wrapper>
)

export default Spinner
