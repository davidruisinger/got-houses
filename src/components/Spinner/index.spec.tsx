import 'jest-styled-components'
import React from 'react'
import testRenderer from 'react-test-renderer'
import Spinner from './index'

const tree = testRenderer.create(<Spinner />).toJSON()

describe('<Spinner />', () => {
  it('matches the snapshot', () => {
    expect(tree).toMatchSnapshot()
  })
})
