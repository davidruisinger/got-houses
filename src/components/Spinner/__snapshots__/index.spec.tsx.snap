// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> matches the snapshot 1`] = `
.c0 {
  width: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  height: calc(6px * 3);
  padding: 3em;
  box-sizing: border-box;
}

.c1 {
  border-radius: 50%;
  width: 6px;
  height: 6px;
  margin: 0 3px;
  -webkit-animation: CncwE 1s ease-in-out infinite;
  animation: CncwE 1s ease-in-out infinite;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
}

.c2 {
  border-radius: 50%;
  width: 6px;
  height: 6px;
  margin: 0 3px;
  -webkit-animation: CncwE 1s ease-in-out infinite;
  animation: CncwE 1s ease-in-out infinite;
  -webkit-animation-delay: -0.9s;
  animation-delay: -0.9s;
}

.c3 {
  border-radius: 50%;
  width: 6px;
  height: 6px;
  margin: 0 3px;
  -webkit-animation: CncwE 1s ease-in-out infinite;
  animation: CncwE 1s ease-in-out infinite;
  -webkit-animation-delay: -0.8s;
  animation-delay: -0.8s;
}

<div
  className="c0"
  direction="row"
  wrap="nowrap"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
</div>
`;
